File: Accenture/7.File_X/code.py
def latest_version(S):
    latest = -1 
    for file in S:
        parts = file.split('_')
        if len(parts) == 2 and parts[0] == "File" and parts[1].isdigit():
            version = int(parts[1]) 
            latest = max(latest, version) 
    return latest
n = int(input("Enter number of files: "))
S = []
for i in range(n):
    S.append(input(f"Enter file name {i+1}: "))
print("Latest version:", latest_version(S))

File: Accenture/7.File_X/code.cpp
#include <bits/stdc++.h>
using namespace std;
bool isNumber(string& s) {
    for (char c : s) {
        if (!isdigit(c)) return false;}
    return true;}
int latestVersion(vector<string>& S) {
    int latest = -1;
    for (auto& file : S) {
        int underscorePos = file.find('_');
        if (underscorePos != string::npos) { 
            string part1 = file.substr(0, underscorePos);
            string part2 = file.substr(underscorePos + 1);

            if (part1 == "File" && isNumber(part2)) { 
                int version = stoi(part2); 
                latest = max(latest, version); }}}
    return latest;}
int main() {
    int n;
    cin >> n;
    vector<string> S(n);
    for (int i = 0; i < n; i++) {
        cin >> S[i];}
    int latest = latestVersion(S);
    cout << "Latest version: " << latest << endl;
    return 0;}
